pkg/sql/create_index.go

```diff
@@ -141,7 +141,7 @@ func (p *planner) setupFamilyAndConstraintForShard(
 		if err != nil {
 			return err
 		}
-		ck.Validity = descpb.ConstraintValidity_Validating
+		ck.Validity = descpb.ConstraintValidity_Validated
 		tableDesc.AddCheckMutation(ck, descpb.DescriptorMutation_ADD)
```

Comment:
This doesn't quite work due to some invariant checking in the tests, as you'll see. I think there's a few option here. One is to make this work. Another is to have the code in the schema changer detect the lack of need to validate this. My preference is for the latter. There's a question of how to validate that exactly. There's a variety of sound approaches. One reasonable one could be to appeal to the code which produces the expression and constraint and just match them that way. Another, more decoupled, but likely more bug prone way, would be to write some code which says `mod(fnv32(...), N) in (0..N-1)` is valid. I think there'd be some edge cases with `NULL` to think about for that.